src/care/forall.h

```diff
@@ -319,21 +324,19 @@ namespace care {
    ///
    ////////////////////////////////////////////////////////////////////////////////
    template <typename LB>
-   void forall(raja_chai_everywhere, const char * fileName, int lineNumber,
+   void forall(managed_ptr_update, const char * fileName, int lineNumber,
                int start, const int end, LB body) {
       // preLoopPrint and postLoopPrint are handled in this call.
       forall(RAJA::seq_exec{}, fileName, lineNumber, start, end, body);
 
-#if defined(CARE_GPUCC) || CARE_ENABLE_GPU_SIMULATION_MODE
+#if defined(CARE_GPUCC)
       const int length = end - start;
 
       if (length != 0) {
          chai::ArrayManager* threadRM = chai::ArrayManager::getInstance();
          threadRM->setExecutionSpace(chai::GPU);
 
-#if CARE_ENABLE_GPU_SIMULATION_MODE
-         RAJA::forall<RAJA::seq_exec>(RAJA::RangeSegment(start, end), body);
-#elif defined(__CUDACC__)
+#if defined(__CUDACC__)
          RAJA::forall< RAJA::cuda_exec<CARE_CUDA_BLOCK_SIZE, CARE_CUDA_ASYNC>>(RAJA::RangeSegment(start, end), body);
 #elif defined(__HIPCC__)
          RAJA::forall< RAJA::hip_exec<CARE_CUDA_BLOCK_SIZE, CARE_CUDA_ASYNC>>(RAJA::RangeSegment(start, end), body);
```

Comment:
This is not at all related to your PR at all, but looking back at this we probably should not be using a variable called CARE_CUDA_BLOCK_SIZE for both CUDA and HIP . I don't know if the right change is to re-name this for CARE_GPU_BLOCK_SIZE, or to make two different variables in case a different number os optimal for each type of GPU.

Also, we probably should be using RAJADeviceExec (which is equal to RAJA::cuda_exec or RAJA::hip_exec or RAJA::omp_parallel_for_exec depending on how we compile). I'll add issues about these things